src/utils/Api.js

```diff
@@ -0,0 +1,130 @@
+class Api {
+    constructor({ baseUrl, headers }) {
+        this._baseUrl = baseUrl;
+        this._headers = headers;
+    }
+
+    _checkResponse(res) {
+        if (res.ok) {
+            return res.json();
+        }
+        return Promise.reject(`Error: ${res.status}`);
+    }
+
+    getAppInfo() {
+        return Promise.all([this.getInitialCards(), this.getUserInfo()]).then(
+            ([cards, userData]) => {
+                return [cards, userData];
+            }
+        );
+    }
+
+    getInitialCards() {
+        return fetch(`${this._baseUrl}/cards`, {
+            headers: this._headers,
+        }).then((res) => {
+            if (res.ok) {
+                return res.json();
+            }
+            return Promise.reject(`Error: ${res.status}`);
+        });
+    }
+
+    getUserInfo() {
+        return fetch(`${this._baseUrl}/users/me`, {
+            headers: this._headers,
+        }).then((res) => {
+            if (res.ok) {
+                return res.json();
+            }
+            return Promise.reject(`Error: ${res.status}`);
```

Comment:
Please, reuse the `_checkResponse` method everywhere, to avoid code duplication


```js

    getInitialCards() {
        return fetch(`${this._baseUrl}/cards`, {
            headers: this._headers,
        }).then(this._checkResponse);  // like this
    }

```